plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -1592,6 +1600,45 @@ private ColumnMapping uuidColumnMapping()
                 uuidWriteFunction());
     }
 
+    private static ColumnMapping vectorColumnMapping(ConnectorSession session)
+    {
+        return ColumnMapping.objectMapping(
+                new ArrayType(REAL),
+                vectorReadFunction(),
+                vectorWriteFunction(session),
+                DISABLE_PUSHDOWN);
+    }
+
+    private static ObjectReadFunction vectorReadFunction()
+    {
+        return ObjectReadFunction.of(Block.class, (resultSet, columnIndex) -> {
+            // getArray is unsupported for vectors type
+            String result = resultSet.getString(columnIndex);
+            if (result == null) {
+                BlockBuilder builder = REAL.createBlockBuilder(null, 10);
```

Comment:
Instead of `REAL` would `Decimal` work here ? Since NaN and Infinity are not supported ?

Reply:
The expected precision and scale are unclear for me. 
> Each vector takes 4 * dimensions + 8 bytes of storage. Each element is a single-precision floating-point number (like the real type in Postgres), and all elements must be finite (no NaN, Infinity or -Infinity). Vectors can have up to 16,000 dimensions.

* https://github.com/pgvector/pgvector?tab=readme-ov-file#vector-type
* https://github.com/pgvector/pgvector-java